### PR TITLE
ESD-1323: add unit tests for ix and nat_gateway inputs, outputs, and prompts

### PR DESCRIPTION
**Pull request by Phil-Browne**
Fills unit test gaps in the `ix` and `nat_gateway` packages identified in ESD-1323.

- `ix`: adds `ix_inputs_test.go` (JSON edge cases, flag no-ops), `ix_output_test.go` (XML format, field mapping, nil safety), `ix_prompts_test.go` (error paths on each prompt beyond the first)
- `nat_gateway`: adds `nat_gateway_inputs_test.go` (flag and JSON paths for create/update, explicit-field tracking), `nat_gateway_output_test.go` (all four formats for gateways/sessions/telemetry/validate/buy, nil-safety), `nat_gateway_prompts_test.go` (create and update happy paths plus error cases)

80 new tests total.

### CODE REVIEW

File: internal/commands/nat_gateway/nat_gateway_prompts_test.go [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+
+func TestPromptForCreateNATGatewayDetails_EmptyName(t *testing.T) {
+	origPrompt := utils.GetResourcePrompt()
+	defer utils.SetResourcePrompt(origPrompt)
+
+	utils.SetResourcePrompt(mockNGPromptSequence([]string{""}))
+
+	_, err := promptForCreateNATGatewayDetails(true)
+	assert.Error(t, err)
+}
```
[the remark this answers is not in the document]
Comment (Phil-Browne) [reply internal/commands/nat_gateway/nat_gateway_prompts_test.go:101]: Fixed — the test now provides valid responses for all 9 prompts so the function reaches `ValidateCreateNATGatewayRequest`, which correctly returns an error for an empty name. Added an `assert.Contains(err.Error(), "name")` assertion to pin the exact failure.

File: internal/commands/nat_gateway/nat_gateway_output_test.go [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+	assert.Contains(t, out, "<items>")
+	assert.Contains(t, out, "ng-1")
+}
+
+func TestPrintNATGateways_Empty(t *testing.T) {
+	for _, format := range []string{"table", "json", "csv", "xml"} {
+		err := printNATGateways([]*megaport.NATGateway{}, format, true)
+		assert.NoError(t, err)
+	}
+}
```
[the remark this answers is not in the document]
Comment (Phil-Browne) [reply internal/commands/nat_gateway/nat_gateway_output_test.go:73]: Fixed — wrapped in `output.CaptureOutput` to suppress stdout during test runs while keeping the no-error assertion.

File: internal/commands/nat_gateway/nat_gateway_output_test.go [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+	})
+	assert.Contains(t, out, "<items>")
+}
+
+func TestPrintNATGatewaySessions_Empty(t *testing.T) {
+	for _, format := range []string{"table", "json", "csv", "xml"} {
+		err := printNATGatewaySessions([]*megaport.NATGatewaySession{}, format, true)
+		assert.NoError(t, err)
+	}
+}
```
[the remark this answers is not in the document]
Comment (Phil-Browne) [reply internal/commands/nat_gateway/nat_gateway_output_test.go:114]: Fixed — same as above, now wrapped in `output.CaptureOutput`.

File: internal/commands/nat_gateway/nat_gateway_inputs_test.go [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			req, explicit, err := processJSONUpdateNATGatewayInput(tt.json, "", "uid-bool")
+			require.NoError(t, err)
+			assert.Equal(t, tt.wantAutoRenew, explicit.AutoRenewTerm)
+			assert.Equal(t, tt.wantBGPShutdown, explicit.BGPShutdownDefault)
+			if tt.wantAutoRenew {
+				assert.Equal(t, tt.autoRenewVal, req.AutoRenewTerm)
+			}
```
[the remark this answers is not in the document]
Comment (Phil-Browne) [reply internal/commands/nat_gateway/nat_gateway_inputs_test.go:247]: Fixed — added `if tt.wantBGPShutdown { assert.Equal(t, tt.bgpShutdownVal, req.Config.BGPShutdownDefault) }` to mirror the existing autoRenew assertion. The `bgpShutdownVal` field is now exercised.

**Review by penzeliz-megaport — APPROVED** [review]
Strong unit coverage for ix and nat_gateway. The per-prompt error sentinel lets each test pinpoint which prompt fails rather than just asserting "no error", and the explicit-field-tracking tests cover the false/zero update path that's easy to get wrong. LGTM.
